frame/identity/src/lib.rs

```diff
@@ -474,18 +474,28 @@ decl_module! {
 		/// - One storage mutation (codec `O(R)`).
 		/// - One event.
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(MINIMUM_WEIGHT)]
-		fn add_registrar(origin, account: T::AccountId) {
+		#[weight = FunctionOf(
+			|(_, &old_count): (&T::AccountId, &u32)| {
+				T::DbWeight::get().reads_writes(1, 1)
+				+ 500_000 * (old_count as u64)
```

Comment:
```suggestion
				+ 500_000 * (old_count as Weight)
```

Better to use the `Weight` type so if things change again, less code needs to be updated